src/constraints.jl

```diff
@@ -562,60 +567,82 @@ into account in the primal value of the constraint. For instance, the constraint
-        <:MOI.AbstractVectorFunction, <:MOI.AbstractVectorSet}})::Vector{Float64}
-    return MOI.get(con_ref.model, MOI.ConstraintPrimal(), con_ref)
+    con_ref::ConstraintRef{
+        Model, <:_MOICON{<:MOI.AbstractVectorFunction, <:MOI.AbstractVectorSet}
+    },
+    result
+)::Vector{Float64}
+    return MOI.get(con_ref.model, MOI.ConstraintPrimal(result), con_ref)
 end
 
 """
-    has_duals(model::Model)
+    has_duals(model::Model; result::Int = 1)
 
-Return true if the solver has a dual solution available to query, otherwise
-return false.
+Return `true` if the solver has a dual solution in result index `result`
+available to query, otherwise return `false`.
 
-See also [`dual`](@ref) and [`shadow_price`](@ref).
+See also [`dual`](@ref), [`shadow_price`](@ref), and [`result_count`](@ref).
 """
-has_duals(model::Model) = dual_status(model) != MOI.NO_SOLUTION
+function has_duals(model::Model; result::Int = 1)
+    return dual_status(model; result = result) != MOI.NO_SOLUTION
+end
 
 """
-    dual(con_ref::ConstraintRef)
+    dual(con_ref::ConstraintRef; result::Int = 1)
+
+Return the dual value of constriant `con_ref` associated with result index
+`result` of the most-recent solution returned by the solver.
 
-Get the dual value of this constraint in the result returned by a solver.
 Use `has_dual` to check if a result exists before asking for values.
-See also [`shadow_price`](@ref).
+
+See also: [`result_count`](@ref), [`shadow_price`](@ref).
 """
-function dual(con_ref::ConstraintRef{Model, <:_MOICON})
-    return reshape_vector(_constraint_dual(con_ref), dual_shape(con_ref.shape))
+function dual(con_ref::ConstraintRef{Model, <:_MOICON}; result::Int = 1)
+    return reshape_vector(
+        _constraint_dual(con_ref, result),
+        dual_shape(con_ref.shape)
+    )
 end
 
 # Returns the value of MOI.ConstraintPrimal in a type-stable way
 function _constraint_dual(
-    con_ref::ConstraintRef{Model, <:_MOICON{
-        <:MOI.AbstractScalarFunction, <:MOI.AbstractScalarSet}})::Float64
-    return MOI.get(con_ref.model, MOI.ConstraintDual(), con_ref)
+    con_ref::ConstraintRef{
+        Model, <:_MOICON{<:MOI.AbstractScalarFunction, <:MOI.AbstractScalarSet}
+    },
+    result::Int
+)::Float64
+    return MOI.get(con_ref.model, MOI.ConstraintDual(result), con_ref)
 end
 function _constraint_dual(
-    con_ref::ConstraintRef{Model, <:_MOICON{
-        <:MOI.AbstractVectorFunction, <:MOI.AbstractVectorSet}})::Vector{Float64}
-    return MOI.get(con_ref.model, MOI.ConstraintDual(), con_ref)
+    con_ref::ConstraintRef{
+        Model, <:_MOICON{<:MOI.AbstractVectorFunction, <:MOI.AbstractVectorSet}
+    },
+    result::Int
+)::Vector{Float64}
+    return MOI.get(con_ref.model, MOI.ConstraintDual(result), con_ref)
 end
 
 
 """
-    shadow_price(con_ref::ConstraintRef)
+    shadow_price(con_ref::ConstraintRef; result::Int = 1)
+
+Return the change in the objective from an infinitesimal relaxation of the
+constraint.
```

Comment:
`result` is not used in the docstring

Reply:
It's also not clear if we need to expose `result` from `shadow_price`. I would be surprised if anyone ever asks for it. It may be better to keep this interface as simple as possible because it's for teaching use only.